modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/bindings/GML2EncodingUtils.java

```diff
@@ -128,29 +128,6 @@ public static String toURI(CoordinateReferenceSystem crs, SrsSyntax srsSyntax) {
         return null;
     }
 
-    /**
-     * Determines the crs of the geometry by checking {@link Geometry#getUserData()}.
-     *
-     * <p>This method returns <code>null</code> when no crs can be found.
-     */
-    public static CoordinateReferenceSystem getCRS(Geometry g) {
```

Comment:
Direct removal of public API is not acceptable, the process is:
- Provide an explanation as to why the method is problematic in the PR.
- Deprecate
- Eventually provide instructions in the upgrade guide

In this case, it seems there has been a centralization of the method to ``gt-main`` new ``GeometryUtil``, and in a similar manner, centralization of duplicated code to set the CRS in a Geometry. Which is a good thing!
In this case you should still keep the original method, deprecated it, and delegate to ``GeometryUtil`` instead.

Also wondering if the existing ``JTS`` class could not be a suitable target for that method, rather than creating a new utility class (it's already dealing with CRS objects for other reasons).
@jodygarnett might have an opinion on this.

Reply:
Okay. JTS seems indeed be a good choice.